src/app/dataset-view/dataset.component.ts

```diff
@@ -114,22 +107,12 @@ export class DatasetComponent implements OnInit, AfterContentInit, OnDestroy {
             },
         ];
 
-    /* eslint-disable  @typescript-eslint/no-explicit-any */
-    public tableData: {
-        isTableHeader: boolean;
-        displayedColumns?: any[];
-        tableSource: any;
-        latestMetadataBlock?: MetadataBlockFragment;
-        isResultQuantity: boolean;
-        isClickableRow: boolean;
-        pageInfo: PageInfoInterface;
-        totalCount: number;
-    };
-    public searchData:
-        | SearchOverviewDatasetsInterface[]
-        | SearchHistoryInterface[] = [];
-
-    public datasetMetadata: {};
+    public tableData: DatasetViewContentInterface;
+    public datasetHistory: MetadataBlockFragment[];
+    private searchOverviewDatasets: SearchHistoryInterface[] = [];
+    private searchDataDatasets: SearchHistoryInterface[] = [];
+    private searchMetadata: SearchOverviewDatasetsInterface[] = [];
```

Comment:
So previously we had two fields (`tableData` and `searchData`) that were reused between multiple instances of tables. Messy, but understandable.

Now we have: 
* Individual fields per each table instance (history, metadata, sample data, query results) - great!
* ... BUT ALSO `this.tableData` field which seems to be still shared reused between all 4 instances of tables
* ... AND ALSO `this.tableData.datasetOverviewSource`, `this.tableData.datasetHistorySource`, ... which are assigned from `this.datasetOverviewSource` and other respective fields ... so a separate data within shared data?

This doesn't look like a completed separation of state.